clients/src/main/java/org/apache/kafka/clients/producer/TransactionState.java

```diff
@@ -108,6 +378,15 @@ public synchronized void setPidAndEpoch(long pid, short epoch) {
      * messages will return an OutOfOrderSequenceException.
      */
     public synchronized void resetProducerId() {
+        if (isTransactional()) {
```

Comment:
In the javadoc above, `it's` -> `its`. Also, don't we have to update it given the change to the implementation?